plugins/webapi/faucet/plugin.go

```diff
@@ -25,58 +25,52 @@ var (
+	Tangle       *tangle.Tangle
 }
 
 // Plugin gets the plugin instance.
 func init() {
-	Plugin = node.NewPlugin("WebAPIFaucetEndpoint", deps, node.Enabled, configure)
+	Plugin = node.NewPlugin("WebAPIFaucetEndpoint", deps, node.Disabled, configure)
 }
 
 func configure(_ *node.Plugin) {
-	deps.Server.POST("faucet", requestFunds)
+	deps.Server.POST("faucet", processFaucetRequest)
 }
 
-// requestFunds creates a faucet request (0-value) message with the given destination address and
-// broadcasts it to the node's neighbors. It returns the message ID if successful.
-func requestFunds(c echo.Context) error {
+// processFaucetRequest processes the faucet request received via the web API.
+func processFaucetRequest(c echo.Context) error {
 	var request jsonmodels.FaucetRequest
 	if err := c.Bind(&request); err != nil {
 		Plugin.LogInfo(err.Error())
-		return c.JSON(http.StatusBadRequest, jsonmodels.FaucetResponse{Error: err.Error()})
+		return c.JSON(http.StatusBadRequest, jsonmodels.FaucetAPIResponse{Error: err.Error()})
 	}
 
-	Plugin.LogInfo("Received - address:", request.Address)
+	Plugin.LogInfo("Received faucet request via web API - address:", request.Address)
 	Plugin.LogDebug(request)
 
 	addr, err := devnetvm.AddressFromBase58EncodedString(request.Address)
 	if err != nil {
-		return c.JSON(http.StatusBadRequest, jsonmodels.FaucetResponse{Error: "Invalid address"})
+		return c.JSON(http.StatusBadRequest, jsonmodels.FaucetRequestResponse{Error: "Invalid address"})
 	}
 
 	var accessManaPledgeID identity.ID
 	var consensusManaPledgeID identity.ID
-	if request.AccessManaPledgeID != "" {
-		accessManaPledgeID, err = mana.IDFromStr(request.AccessManaPledgeID)
-		if err != nil {
-			return c.JSON(http.StatusBadRequest, jsonmodels.FaucetResponse{Error: "Invalid access mana node ID"})
-		}
+	if request.AccessManaPledgeID == "" {
+		return c.JSON(http.StatusBadRequest, jsonmodels.FaucetAPIResponse{Error: "Invalid access mana node ID"})
 	}
 
-	if request.ConsensusManaPledgeID != "" {
-		consensusManaPledgeID, err = mana.IDFromStr(request.ConsensusManaPledgeID)
-		if err != nil {
-			return c.JSON(http.StatusBadRequest, jsonmodels.FaucetResponse{Error: "Invalid consensus mana node ID"})
-		}
+	if request.ConsensusManaPledgeID == "" {
+		return c.JSON(http.StatusBadRequest, jsonmodels.FaucetAPIResponse{Error: "Invalid consensus mana node ID"})
 	}
 
+	consensusManaPledgeID, err = mana.IDFromStr(request.ConsensusManaPledgeID)
+	accessManaPledgeID, err = mana.IDFromStr(request.AccessManaPledgeID)
+
 	faucetPayload := faucetpkg.NewRequest(addr, accessManaPledgeID, consensusManaPledgeID, request.Nonce)
 
-	msg, err := deps.Tangle.MessageFactory.IssuePayload(faucetPayload)
-	if err != nil {
-		return c.JSON(http.StatusInternalServerError, jsonmodels.FaucetResponse{Error: fmt.Sprintf("Failed to send faucetrequest: %s", err.Error())})
-	}
+	deps.FaucetEvents.WebAPIFaucetRequest.Trigger(&faucet.FaucetRequestEvent{Request: faucetPayload})
```

Comment:
I think it would simplify the code quite a bit and make the API more expressive if you just call `faucet.onWebAPIRequest` directly. The method could then also return an error that you can return via the API (e.g. PoW is not enough, blacklisted) so the user can see the problem.